app/main/helpers/diff_tools.py

```diff
@@ -0,0 +1,276 @@
+import difflib
+from abc import ABCMeta, abstractmethod
+from flask import Markup, escape
+
+
+class BaseDiffTool(object):
+    """
+    Creates a line-by-line diff from two lists of items
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def __init__(self, revision_1, revision_2, if_unchanged):
+        # revisions are lists of strings to diff against each other
+        self.revision_1 = revision_1
+        self.revision_2 = revision_2
+        self.include_unchanged_lines_in_output = if_unchanged
+        self.lines = self._get_lines()
+
+    def render_lines(self):
+        """
+        Returns rendered HTML lines to be displayed in a diff page
+        """
+
+        lines = {
+            'revision_1': [],
+            'revision_2': []
+        }
+        for index, line in enumerate(self.lines['revision_1']):
+            lines['revision_1'].append(
+                Markup(self._render_words_html(
+                    self.lines['revision_1'][index], index+1)
+                )
+            )
+            lines['revision_2'].append(
+                Markup(self._render_words_html(
+                    self.lines['revision_2'][index], index+1)
+                )
+            )
+
+        return lines
+
+    def _get_lines(self):
+        """
+        Turns a pair of input lines into a list of word diffs
+
+        In:
+        self.revision_1: ['Hi there', 'Less letters']
+        self.revision_2: ['Hi there!', 'Less let']
+
+        Out:
+        self.lines: { 'revision_1': ['  Hi', '- there'], ['  Less', '- letters'],
+                      'revision_2': ['  Hi', '+ there!'], ['  Less', '+ let'] }
+        """
+
+        lines = {
+            'revision_1': [],
+            'revision_2': []
+        }
+
+        # Make sure lists of strings are same length
+        # Pad the shorter list with empty strings if necessary
+        # Also means pair of lists in returned dictionary are of equal length
+        revision_1, revision_2 = self._pad_lists_to_max_length(
```

Comment:
You could use [itertools.izip_longest](https://docs.python.org/2/library/itertools.html#itertools.izip_longest) instead of a padding method